tests/ui/pattern/usefulness/consts-opaque.rs

```diff
@@ -124,8 +125,16 @@ fn main() {
 
     match WRAPQUUX {
         Wrap(_) => {}
-        WRAPQUUX => {} // detected unreachable because we do inspect the `Wrap` layer
-        //~^ ERROR unreachable pattern
+        WRAPQUUX => {}
+    }
```

Comment:
This doesn't emit an `unreachable_pattern` lint anymore, as we just treat the constant opaquely. Our exhaustiveness checks check for duplicate or already covered patterns, they don't realize that all patterns have already been covered and thus all remaining ones must be unreachable.

[self-contained playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=6bbc069766dfee8ecfb4e95ecaeac2da)